pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -45,7 +45,7 @@
   private ReadMode _readMode = ReadMode.DEFAULT_MODE;
   private List<String> _sortedColumns = Collections.emptyList();
   private Set<String> _invertedIndexColumns = new HashSet<>();
-  private Set<String> _textIndexColumns = new HashSet<>();
+  private Map<String, Map<String, String>> _textIndexColumns = new HashMap<>();
```

Comment:
Don't put the whole map here. The map contains all the properties, not for text column only. I think you can keep this config unchanged, but check the field config when loading the text index.

Reply:
Done. FieldConfig contains a properties map Map<String, String> per column